#! /bin/sh /usr/share/dpatch/dpatch-run
## 07-programming-error-unicode.dpatch by  <gilir@ubuntu.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: upstream bug : http://bugzilla.gnome.org/show_bug.cgi?id=583823
## DP: ubuntu bug : https://bugs.launchpad.net/ubuntu/+source/conduit/+bug/371175

@DPATCH@
diff -urNad conduit~/conduit/MappingDB.py conduit/conduit/MappingDB.py
--- conduit~/conduit/MappingDB.py	2009-08-30 13:33:34.000000000 +0200
+++ conduit/conduit/MappingDB.py	2009-08-30 13:40:49.000000000 +0200
@@ -38,8 +38,8 @@
         self.sinkRid = rid
         
     def values(self):
-        return (self.sourceUID,self.sourceRid.get_UID(),self.sourceRid.get_mtime(),self.sourceRid.get_hash(),
-                self.sinkUID,self.sinkRid.get_UID(),self.sinkRid.get_mtime(),self.sinkRid.get_hash())
+        return (unicode(self.sourceUID),unicode(self.sourceRid.get_UID()),self.sourceRid.get_mtime(),self.sourceRid.get_hash(),
+                unicode(self.sinkUID),unicode(self.sinkRid.get_UID()),self.sinkRid.get_mtime(),self.sinkRid.get_hash())
         
 class MappingDB:
     """
